mubench.reviewsite/tests/ReviewControllerTest.php

```diff
@@ -54,4 +54,60 @@ function test_stores_violation_types()
         self::assertEquals("missing/call", $violation_types[0]["name"]);
     }
 
+    function test_determine_next_misuse()
+    {
+        $this->setupSomeRunWithThreeMisusesAndTwoReviews();
+
+        list($previous_misuse, $next_misuse, $next_reviewable_misuse, $misuse)  = $this->reviewController->determineNavigationTargets(Misuse::all(), '-p-', '-v-', '0', Reviewer::find(1));
+        self::assertEquals('2', $previous_misuse->misuse_muid);
+        self::assertEquals('1', $next_misuse->misuse_muid);
+        self::assertEquals('2', $next_reviewable_misuse->misuse_muid);
+        self::assertEquals('0', $misuse->misuse_muid);
+    }
+
+    function test_determine_next_is_previous_misuse()
+    {
+        $this->setupSomeRunWithThreeMisusesAndTwoReviews();
+
+        list($previous_misuse, $next_misuse, $next_reviewable_misuse, $misuse)  = $this->reviewController->determineNavigationTargets(Misuse::all(), '-p-', '-v-', '2', Reviewer::find(1));
+        self::assertEquals('1', $previous_misuse->misuse_muid);
+        self::assertEquals('1', $next_misuse->misuse_muid);
+        self::assertNull($next_reviewable_misuse);
+        self::assertEquals('2', $misuse->misuse_muid);
+    }
+
+    private function setupSomeRunWithThreeMisusesAndTwoReviews()
+    {
+        $runsController = new RunsController($this->container);
+        $runsController->addRun(1, 'test-detector', '-p-', '-v-',json_decode(json_encode(
+            [
+                "result" => "success",
+                "runtime" => 42.1,
+                "number_of_findings" => 23,
+                "timestamp" => 12,
+                "potential_hits" => [
+                    [
+                        "misuse" => "0",
+                        "rank" => 0,
+                    ],
+                    [
+                        "misuse" => "1",
+                        "rank" => 0,
+                    ],
+                    [
+                        "misuse" => "2",
+                        "rank" => 0,
+                    ]
+                ]
+            ]
+        )));
+        Misuse::create(['misuse_muid' => '1', 'run_id' => 1, 'detector_id' => 1]);
+        Misuse::create(['misuse_muid' => '2', 'run_id' => 1, 'detector_id' => 1]);
+        Reviewer::create(['name' => 'reviewer2']);
```

Comment:
Please save the reviewer(s) in variables and pass their ids on, otherwise it is hard to track which reviewer has which id and you're relying on the database generating keys in insertion order!